include/deal.II/fe/fe_tools.h

```diff
@@ -977,6 +977,37 @@ namespace FETools
                           const bool do_tensor_product = true);
 
     /**
+     * Same as above for an arbitrary number of parameters of type
+     * <code>std::pair<std::unique_ptr<FiniteElement<dim, spacedim>>, unsigned int></code>
+     * and <code>do_tensor_product = true</code>.
+     */
+    template <int dim, int spacedim, class... FESystems,
+              typename = typename std::enable_if<all_same_as<typename std::decay<FESystems>::type...,
+                                                             std::pair<std::unique_ptr<FiniteElement<dim, spacedim>>,
+                                                                 unsigned int>>::value>::type>
+    FiniteElementData<dim>
+    multiply_dof_numbers (FESystems &&... fe_systems)
+    {
```

Comment:
would you be willing to move the implementation of the function to the bottom of the file, leaving the declaration here? same for the other new functions here